arangod/Cluster/ClusterTrxMethods.cpp

```diff
@@ -225,13 +225,16 @@ Future<Result> commitAbortTransaction(arangodb::TransactionState* state,
                   ServerState::instance()->getId());
   }
 
+  char const* stateString = nullptr;
   fuerte::RestVerb verb;
   if (status == transaction::Status::COMMITTED) {
+    stateString = "commit";
     verb = fuerte::RestVerb::Put;
   } else if (status == transaction::Status::ABORTED) {
+    stateString = "abort";
     verb = fuerte::RestVerb::Delete;
   } else {
-    TRI_ASSERT(false);
+    THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_INTERNAL, "invalid state for commit/abort operation");
```

Comment:
```suggestion
     TRI_ASSERT(false);
    THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_INTERNAL, "invalid state for commit/abort operation");
```

I would like to keep the Assert, and include the THROW for production.
This is illegal state and should not somehow slip through our Jenkins, just because the tests silently ignore errors on APIs they are not testing.